From 1be9450653513cf19cf910b0e3cbcea01c57fed2 Mon Sep 17 00:00:00 2001
From: Allen Porter <allen@thebends.org>
Date: Wed, 19 Jan 2022 20:37:41 -0800
Subject: [PATCH] Make basic auth optional for private handler

Make basic auth only enabled when explicitly configured, so that RTSPtoWeb can be used easier in environments that provde their own security e.g. network restrictions to disable broad access.
---
 apiHTTPRouter.go | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/apiHTTPRouter.go b/apiHTTPRouter.go
index 372d0e6..dcc3993 100644
--- a/apiHTTPRouter.go
+++ b/apiHTTPRouter.go
@@ -36,7 +36,10 @@ func HTTPAPIServer() {
 
 	public.Use(CrossOrigin())
 	//Add private login password protect methods
-	privat := public.Group("/", gin.BasicAuth(gin.Accounts{Storage.ServerHTTPLogin(): Storage.ServerHTTPPassword()}))
+	privat := public.Group("/")
+	if Storage.ServerHTTPLogin() != "" && Storage.ServerHTTPPassword() != "" {
+		privat.Use(gin.BasicAuth(gin.Accounts{Storage.ServerHTTPLogin(): Storage.ServerHTTPPassword()}))
+	}
 	public.LoadHTMLGlob(Storage.ServerHTTPDir() + "/templates/*")
 
 	/*
-- 
2.25.1

